translations/pt-br/04-ferramentas-utilizadas.md

```diff
@@ -0,0 +1,142 @@
+
+- GitHub & Git
+- CodeSee
+- VS Code (O editor)
+- Discord
+
+## Pré-requisitos
+
+Antes de completar o seguinte passo a passo, complete o seguinte:
+- Crie uma [conta no GitHub](https://github.com/)
+- Baixe o [VS Code](https://code.visualstudio.com/)
+
+## GitHub & Git
+
+### Introdução ao Git
+
+Git é um sistema de controlo de versão distribuído projetado para lidar com projetos pequenos e muito grandes com velocidade e eficiência. Foi criado por Linus Torvalds, o criador do kernel Linux, para gerir o desenvolvimento do próprio kernel Linux.
+
+Git permite que os desenvolvedores acompanhem as alterações no seu código, colaborem com outros desenvolvedores e gerenciem várias versões dos seus projetos. Alguns dos recursos principais do Git incluem:
+
+- **Branch**: Git permite que os desenvolvedores criem ramificações, cópias separadas da sua base de código. Isso permite que eles trabalhem em novos recursos ou correções de bugs sem afetar a base de código principal. Uma vez que as alterações estejam concluídas, eles podem mesclar a ramificação de volta para a base de código principal.
+- **Área de preparação(Stagging area)**: Git fornece uma área de preparação onde os desenvolvedores podem preparar as suas alterações antes de as comprometerem com o repositório. Isso permite que eles revisem e organizem as suas alterações antes de as tornarem permanentes.
+- **Controle de versão distribuído**: Ao contrário dos sistemas de controlo de versão centralizados, o Git é um sistema de controlo de versão distribuído. Isso significa que cada desenvolvedor tem uma cópia completa do repositório na máquina local, permitindo-lhes trabalhar offline e independentemente de um servidor central.
+
+### Introdução ao GitHub
+
+O GitHub é uma plataforma baseada na web para controlo de versão e colaboração que utiliza o Git como tecnologia subjacente. Ele fornece uma interface simples e amigável para gerir repositórios Git, bem como uma série de recursos e ferramentas para colaborar com outros desenvolvedores.
+
+Algumas das principais funcionalidades do GitHub incluem:
+
+- **Rastramento de issues**: O GitHub fornece um sistema integrado de rastreamento de problemas que permite que os desenvolvedores criem, atribuam e gerenciem problemas relacionados aos seus projetos. Os problemas podem ser categorizados usando etiquetas, marcos e pessoas atribuídas, tornando fácil acompanhar o progresso do projeto e priorizar tarefas.
+- **Pull requests**: O GitHub introduz o conceito de pedidos de pull, que permitem que os desenvolvedores proponham alterações à base de código de um projeto. Os pedidos de pull permitem a revisão de código, discussão e colaboração antes que as alterações sejam mescladas na base de código principal.
```

Comment:
```suggestion
- **Pull requests**: O GitHub introduz o conceito de solicitação de pull, que permitem que os desenvolvedores proponham alterações à base de código de um projeto. As solicitação de pull permitem a revisão de código, discussão e colaboração antes que as alterações sejam mescladas na base de código principal.
```
https://docs.github.com/pt/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests
![image](https://github.com/open-sauced/intro/assets/61759797/1242ea52-e5e2-40e3-9d7f-3f22ae10690c)